### PR TITLE
Update localization system

### PR DESCRIPTION
**Pull request by TPGamesNL**
### Description
Changes:
- Removed non-message stuff from non-English localization files, as we decided that non-message stuff shouldn't be translated, since it could break script compatibility between languages.
- Made the language entries fall back to English if the entries aren't present in the localized language file
- Fix some messages not showing up in the right language, even when there where language entries for them
- Completed German language file (thanks to Runakai)

---
**Target Minecraft Versions:** any
**Requirements:** none
**Related Issues:** #136, #1359, #2585, #2890, #3859, #4155, #4233

### CODE REVIEW

**Comment by TPGamesNL** [comment]
I've now added a new language file: `default.lang`. This file is meant for everything that shouldn't be translated (the only things that should be translated are messages). The file `english.lang` is now much more similar to the other languages, since it only contains messages. This makes it a better template for new languages to be added.

You can use the `plugins/Skript/lang/%language%.lang` file to override entries from the language files in the plugin jar (except for the `default.lang`). You don't need to copy the entire language file from the jar either, you only need the entries you want to change.

While technically not a message, I've added the `<none>` entry to the language files (so not in default.lang) as well, simply because it is one of the values people change most often, and it's much easier to change it in the external file (`plugins/Skript/lang`) than in the internal file (plugin jar).

**Comment by TPGamesNL** [comment]
The non-default language files are now copied to `plugins/Skript/lang` by default, so there is no need to look up the keys in the in-jar language files.

File: src/main/java/ch/njol/skript/localization/Language.java [hunk 1]
```diff
@@ -135,132 +104,149 @@ public static String get(final String key) {
...
 	}
 	
-	private static boolean load(final SkriptAddon addon, final String name) {
+	private static boolean load(SkriptAddon addon, String name) {
 		if (addon.getLanguageFileDirectory() == null)
 			return false;
-		final HashMap<String, String> l = load(addon.plugin.getResource(addon.getLanguageFileDirectory() + "/" + name + ".lang"), name);
-		final File f = new File(addon.plugin.getDataFolder(), addon.getLanguageFileDirectory() + File.separator + name + ".lang");
+		// Backwards addon compatibility
+		if (name.equals("english") && addon.plugin.getResource(addon.getLanguageFileDirectory() + "/default.lang") == null)
```
Comment (APickledWalrus) [inline src/main/java/ch/njol/skript/localization/Language.java:241]: Should everything be using `File.separator` in place of just `/`?

**Review by APickledWalrus — COMMENTED** [review]
Just these two things! Looks good to me other than that - another review from someone else is probably needed though :)

File: src/main/java/ch/njol/skript/localization/Language.java (hunk 1 above)
Comment (TPGamesNL) [reply src/main/java/ch/njol/skript/localization/Language.java]: Turns out zip files should always have forward slashes:
image

**Comment by Romitou** [comment]
How about putting the `<none>` notation in the default.lang instead? I think that displaying a null value should not be translated, this is also part of the "core" of Skript from my point of view. For example (even if this practice is bad), some people do: `if "%foobar%" is "<none>":`, etc. In communities other than English, the use of <none> is common and correctly understood, and I think giving the option to change this will cause confusion.

**Comment by TPGamesNL** [comment]
@Romitou I agree that it belongs in default.lang, but this is also a value people tend to change, so I decided to put it in the language file: https://github.com/SkriptLang/Skript/pull/3999#issuecomment-841679971

File: src/main/resources/lang/default.lang [hunk 2]
```diff
@@ -0,0 +1,1853 @@
...
+	hotbar_move_and_readd: hotbar move and readd
+	hotbar_swap: swap with hotbar, swap items with hotbar, hotbar swap, hotbar swap items
+	clone_stack: clone stack
+	collect_to_cursor: collect to cursor, collect items to cursor
+	unknown: unknown, unsupported, custom
+
+# -- Inventory Click Types --
+click types:
+	left: left mouse button, left mouse, LMB
+	shift_left: left mouse button with shift, left mouse with shift, Shift+RMB
```
Comment (AyhamAl-Ali) [inline src/main/resources/lang/default.lang:1621]: Should be Shift+LMB
Related #4233

**Review by AyhamAl-Ali — CHANGES_REQUESTED** [review]

**Pull request merged**